File: SagaSpareParts/Modules/class_Tools.cs
using Microsoft.Win32;
using System;
using System.IO;
using System.Security.Cryptography;
using System.Text;
using System.Text.RegularExpressions;
using System.Windows.Forms;

namespace MyClassLibrary.Classes
{
    public static class class_Tools
    {
        private const string passphrase = "webMh@rjun";

        public static string Encrypt(string sText)
        {
            byte[] Results;
            var UTF8 = new UTF8Encoding();
            var HashProvider = new MD5CryptoServiceProvider();
            var TDESKey = HashProvider.ComputeHash(UTF8.GetBytes(passphrase));
            var TDESAlgorithm = new TripleDESCryptoServiceProvider()
            {
                Key = TDESKey,
                Mode = CipherMode.ECB,
                Padding = PaddingMode.PKCS7
            };
            var DataToEncrypt = UTF8.GetBytes(sText);
            try
            {
                var Encryptor = TDESAlgorithm.CreateEncryptor();
                Results = Encryptor.TransformFinalBlock(DataToEncrypt, 0, DataToEncrypt.Length);
            }
            catch (Exception ex)
            {
                class_Procedures.Show_Error(ex);
                return null;
            }
            finally
            {
                TDESAlgorithm.Clear();
                HashProvider.Clear();
            }

            return Convert.ToBase64String(Results);
        }

        public static string Decrypt(string sText, string sDefault = "")
        {
            if (ReferenceEquals(sText, string.Empty))
                return string.Empty;
            byte[] Results;
            var UTF8 = new UTF8Encoding();
            var HashProvider = new MD5CryptoServiceProvider();
            var TDESKey = HashProvider.ComputeHash(UTF8.GetBytes(passphrase));
            var TDESAlgorithm = new TripleDESCryptoServiceProvider()
            {
                Key = TDESKey,
                Mode = CipherMode.ECB,
                Padding = PaddingMode.PKCS7
            };
            try
            {
                var DataToDecrypt = Convert.FromBase64String(sText);
                var Decryptor = TDESAlgorithm.CreateDecryptor();
                Results = Decryptor.TransformFinalBlock(DataToDecrypt, 0, DataToDecrypt.Length);
            }
            catch (Exception ex)
            {
                class_Procedures.Show_Error(ex);
                return sDefault;
            }
            finally
            {
                TDESAlgorithm.Clear();
                HashProvider.Clear();
            }

            return UTF8.GetString(Results);
        }

        public static void ReplaceInFile(string OrigFile, string NewFile, string searchText, string replaceText)
        {
            var reader = new StreamReader(OrigFile);
            string content = reader.ReadToEnd();
            reader.Close();
            content = Regex.Replace(content, searchText, replaceText);
            using (var writer = new StreamWriter(NewFile))
            {
                writer.Write(content);
            }
        }

        public static void SubKeyTreeDelete(RegistryKey registryPath, string keyToDelete)
        {
            if (registryPath is object && !string.IsNullOrWhiteSpace(keyToDelete))
            {
                bool throwOnMissingSubKey = false;
                try
                {
                    registryPath.DeleteSubKeyTree(keyToDelete, throwOnMissingSubKey);
                }
                catch
                {
                }
            }
        }

        public static string RegKeyGet(string keySection, string keyName, string keyDefault)
        {
            string sPath = $@"{Application.ProductName}\\{keySection}\\{keyName}";
            var key = Registry.CurrentUser.OpenSubKey(sPath, true);
            if (key is null)
            {
                key = Registry.CurrentUser.CreateSubKey(sPath);
                key.SetValue(keyName, keyDefault);
            }

            return Convert.ToString(key.GetValue(keyName));
        }

        public static bool RegKeyGet(string keySection, string keyName, bool keyDefault)
        {
            string sPath = $@"{Application.ProductName}\\{keySection}\\{keyName}";
            var key = Registry.CurrentUser.OpenSubKey(sPath, true);
            if (key is null)
            {
                key = Registry.CurrentUser.CreateSubKey(sPath);
                key.SetValue(keyName, keyDefault);
            }

            return Convert.ToBoolean(key.GetValue(keyName));
        }

        public static int RegKeyGet(string keySection, string keyName, int keyDefault)
        {
            string sPath = $@"{Application.ProductName}\\{keySection}\\{keyName}";
            var key = Registry.CurrentUser.OpenSubKey(sPath, true);
            if (key is null)
            {
                key = Registry.CurrentUser.CreateSubKey(sPath);
                key.SetValue(keyName, keyDefault);
            }

            return Convert.ToInt32(key.GetValue(keyName));
        }

        public static decimal RegKeyGet(string keySection, string keyName, decimal keyDefault)
        {
            string sPath = $@"{Application.ProductName}\\{keySection}\\{keyName}";
            var key = Registry.CurrentUser.OpenSubKey(sPath, true);
            if (key is null)
            {
                key = Registry.CurrentUser.CreateSubKey(sPath);
                key.SetValue(keyName, keyDefault);
            }

            return Convert.ToDecimal(key.GetValue(keyName));
        }

        public static void RegKeySet(string keySection, string keyName, string keyValue)
        {
            if (keyValue is null)
                return;
            string sPath = $@"{Application.ProductName}\\{keySection}\\{keyName}";
            var key = Registry.CurrentUser.OpenSubKey(sPath, true);
            if (key is object)
            {
                key.SetValue(keyName, keyValue);
                key.Close();
            }
        }

        public static void RegKeySet(string keySection, string keyName, bool keyValue)
        {
            string sPath = $@"{Application.ProductName}\\{keySection}\\{keyName}";
            var key = Registry.CurrentUser.OpenSubKey(sPath, true);
            if (key is object)
            {
                key.SetValue(keyName, keyValue);
                key.Close();
            }
        }

        public static void RegKeySet(string keySection, string keyName, int keyValue)
        {
            string sPath = $@"{Application.ProductName}\\{keySection}\\{keyName}";
            var key = Registry.CurrentUser.OpenSubKey(sPath, true);
            if (key is object)
            {
                key.SetValue(keyName, keyValue);
                key.Close();
            }
        }

        public static void RegKeySet(string keySection, string keyName, decimal keyValue)
        {
            string sPath = $@"{Application.ProductName}\\{keySection}\\{keyName}";
            var key = Registry.CurrentUser.OpenSubKey(sPath, true);
            if (key is object)
            {
                key.SetValue(keyName, keyValue);
                key.Close();
            }
        }
    }
}

File: MyClassLibrary/Classes/class_Tools.vb
Imports Microsoft.Win32
Imports System.IO
Imports System.Security.Cryptography
Imports System.Text
Imports System.Text.RegularExpressions

Namespace Classes
    Public Module class_Tools
        Const passphrase As String = "webMh@rjun"

        Public Function Encrypt(ByVal sText As String) As String
            Dim Results As Byte()
            Dim UTF8 As UTF8Encoding = New UTF8Encoding()
            Dim HashProvider As MD5CryptoServiceProvider = New MD5CryptoServiceProvider()
            Dim TDESKey As Byte() = HashProvider.ComputeHash(UTF8.GetBytes(passphrase))
            Dim TDESAlgorithm As TripleDESCryptoServiceProvider = New TripleDESCryptoServiceProvider With {
                .Key = TDESKey,
                .Mode = CipherMode.ECB,
                .Padding = PaddingMode.PKCS7
            }
            Dim DataToEncrypt As Byte() = UTF8.GetBytes(sText)

            Try
                Dim Encryptor As ICryptoTransform = TDESAlgorithm.CreateEncryptor()
                Results = Encryptor.TransformFinalBlock(DataToEncrypt, 0, DataToEncrypt.Length)
            Catch Ex As CryptographicException
                class_Procedures.Show_Error(Ex)
                Return Nothing
            Finally
                TDESAlgorithm.Clear()
                HashProvider.Clear()
            End Try

            Return Convert.ToBase64String(Results)
        End Function

        Public Function Decrypt(ByVal sText As String, Optional ByVal sDefault As String = "") As String
            If sText Is String.Empty Then Return String.Empty
            Dim Results As Byte()
            Dim UTF8 As UTF8Encoding = New UTF8Encoding()
            Dim HashProvider As MD5CryptoServiceProvider = New MD5CryptoServiceProvider()
            Dim TDESKey As Byte() = HashProvider.ComputeHash(UTF8.GetBytes(passphrase))
            Dim TDESAlgorithm As TripleDESCryptoServiceProvider = New TripleDESCryptoServiceProvider With {
            .Key = TDESKey,
            .Mode = CipherMode.ECB,
            .Padding = PaddingMode.PKCS7
        }

            Try
                Dim DataToDecrypt As Byte() = Convert.FromBase64String(sText)
                Dim Decryptor As ICryptoTransform = TDESAlgorithm.CreateDecryptor()
                Results = Decryptor.TransformFinalBlock(DataToDecrypt, 0, DataToDecrypt.Length)
            Catch Ex As CryptographicException
                class_Procedures.Show_Error(Ex)
                Return sDefault
            Catch Ex As Exception
                class_Procedures.Show_Error(Ex)
                Return sDefault
            Finally
                TDESAlgorithm.Clear()
                HashProvider.Clear()
            End Try

            Return UTF8.GetString(Results)
        End Function

        Public Sub ReplaceInFile(ByVal OrigFile As String, ByVal NewFile As String, ByVal searchText As String, ByVal replaceText As String)
            Dim reader As StreamReader = New StreamReader(OrigFile)
            Dim content As String = reader.ReadToEnd()
            reader.Close()
            content = Regex.Replace(content, searchText, replaceText)

            Using writer As StreamWriter = New StreamWriter(NewFile)
                writer.Write(content)
            End Using
        End Sub

        Public Sub SubKeyTreeDelete(ByVal registryPath As RegistryKey, ByVal keyToDelete As String)
            If registryPath IsNot Nothing AndAlso Not String.IsNullOrWhiteSpace(keyToDelete) Then
                Dim throwOnMissingSubKey As Boolean = False
                Try
                    registryPath.DeleteSubKeyTree(keyToDelete, throwOnMissingSubKey)
                Catch
                End Try
            End If
        End Sub

        Public Function RegKeyGet(ByVal keySection As String, ByVal keyName As String, ByVal keyDefault As String) As String
            Dim sPath As String = $"{Application.ProductName}\\{keySection}\\{keyName}"
            Dim key As RegistryKey = Registry.CurrentUser.OpenSubKey(sPath, True)
            If key Is Nothing Then
                key = Registry.CurrentUser.CreateSubKey(sPath)
                key.SetValue(keyName, keyDefault)
            End If
            Return CStr(key.GetValue(keyName))
        End Function

        Public Function RegKeyGet(ByVal keySection As String, ByVal keyName As String, ByVal keyDefault As Boolean) As Boolean
            Dim sPath As String = $"{Application.ProductName}\\{keySection}\\{keyName}"
            Dim key As RegistryKey = Registry.CurrentUser.OpenSubKey(sPath, True)
            If key Is Nothing Then
                key = Registry.CurrentUser.CreateSubKey(sPath)
                key.SetValue(keyName, keyDefault)
            End If
            Return CBool(key.GetValue(keyName))
        End Function

        Public Function RegKeyGet(ByVal keySection As String, ByVal keyName As String, ByVal keyDefault As Integer) As Integer
            Dim sPath As String = $"{Application.ProductName}\\{keySection}\\{keyName}"
            Dim key As RegistryKey = Registry.CurrentUser.OpenSubKey(sPath, True)
            If key Is Nothing Then
                key = Registry.CurrentUser.CreateSubKey(sPath)
                key.SetValue(keyName, keyDefault)
            End If
            Return CInt(key.GetValue(keyName))
        End Function

        Public Function RegKeyGet(ByVal keySection As String, ByVal keyName As String, ByVal keyDefault As Decimal) As Decimal
            Dim sPath As String = $"{Application.ProductName}\\{keySection}\\{keyName}"
            Dim key As RegistryKey = Registry.CurrentUser.OpenSubKey(sPath, True)
            If key Is Nothing Then
                key = Registry.CurrentUser.CreateSubKey(sPath)
                key.SetValue(keyName, keyDefault)
            End If
            Return CDec(key.GetValue(keyName))
        End Function

        Public Sub RegKeySet(ByVal keySection As String, ByVal keyName As String, ByVal keyValue As String)
            If keyValue Is Nothing Then Return
            Dim sPath As String = $"{Application.ProductName}\\{keySection}\\{keyName}"
            Dim key As RegistryKey = Registry.CurrentUser.OpenSubKey(sPath, True)
            If key IsNot Nothing Then
                key.SetValue(keyName, keyValue)
                key.Close()
            End If
        End Sub

        Public Sub RegKeySet(ByVal keySection As String, ByVal keyName As String, ByVal keyValue As Boolean)
            Dim sPath As String = $"{Application.ProductName}\\{keySection}\\{keyName}"
            Dim key As RegistryKey = Registry.CurrentUser.OpenSubKey(sPath, True)
            If key IsNot Nothing Then
                key.SetValue(keyName, keyValue)
                key.Close()
            End If
        End Sub

        Public Sub RegKeySet(ByVal keySection As String, ByVal keyName As String, ByVal keyValue As Integer)
            Dim sPath As String = $"{Application.ProductName}\\{keySection}\\{keyName}"
            Dim key As RegistryKey = Registry.CurrentUser.OpenSubKey(sPath, True)
            If key IsNot Nothing Then
                key.SetValue(keyName, keyValue)
                key.Close()
            End If
        End Sub

        Public Sub RegKeySet(ByVal keySection As String, ByVal keyName As String, ByVal keyValue As Decimal)
            Dim sPath As String = $"{Application.ProductName}\\{keySection}\\{keyName}"
            Dim key As RegistryKey = Registry.CurrentUser.OpenSubKey(sPath, True)
            If key IsNot Nothing Then
                key.SetValue(keyName, keyValue)
                key.Close()
            End If
        End Sub

    End Module
End Namespace